File: wk6/java/baseballGame.java
// problem link: https://leetcode.com/problems/baseball-game/

class Solution {
    public int calPoints(String[] operations) {
        int ret = 0;
        Stack<Integer> s = new Stack<>();
        s.push(Integer.parseInt(operations[0]));
        ret += s.peek();
        for(int i = 1; i < operations.length; i++) {
            if(operations[i].equals("+")) {
                int a = s.pop(), b = s.peek();
                s.push(a);
                s.push(a + b);
                ret += a + b;
            }
            else if(operations[i].equals("D")) {
                s.push(s.peek() * 2);
                ret += s.peek();
            }
            else if(operations[i].equals("C"))
                ret -= s.pop();
            else {
                s.push(Integer.parseInt(operations[i]));
                ret += s.peek();
            }
        }
        return ret;
    }
}

File: wk6/cpp/baseballGame.cpp
// problem link: https://leetcode.com/problems/baseball-game/description/

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int calPoints(vector<string>& operations) {
        vector<int> v;
        for(auto const &op : operations) {
            if(op[0] == '+')
                v.push_back(v[v.size() - 1] + v[v.size() - 2]);
            else if(op[0] == 'D')
                v.push_back(2 * v[v.size() - 1]);
            else if(op[0] == 'C')
                v.pop_back();
            else
                v.push_back(stoi(op));
        }
        int ret = 0;
        for(auto const &i : v)
            ret += i;
        return ret;
    }
};